From 4089f79312ace19176fd8513f6270ecc4913a4c4 Mon Sep 17 00:00:00 2001
From: shucheng <shucheng@bd-apaas.com>
Date: Tue, 14 Mar 2023 10:12:46 +0800
Subject: [PATCH] mips64le

---
 hack/lib/golang.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index 7e5b2216163..7375026946a 100755
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -25,7 +25,7 @@ readonly KUBE_SUPPORTED_SERVER_PLATFORMS=(
   linux/amd64
   linux/arm
   linux/arm64
-  linux/s390x
+  linux/mips64le
   linux/ppc64le
 )
 
@@ -34,7 +34,7 @@ readonly KUBE_SUPPORTED_NODE_PLATFORMS=(
   linux/amd64
   linux/arm
   linux/arm64
-  linux/s390x
+  linux/mips64le
   linux/ppc64le
   windows/amd64
 )
@@ -46,7 +46,7 @@ readonly KUBE_SUPPORTED_CLIENT_PLATFORMS=(
   linux/386
   linux/arm
   linux/arm64
-  linux/s390x
+  linux/mips64le
   linux/ppc64le
   darwin/amd64
   darwin/arm64
@@ -61,7 +61,7 @@ readonly KUBE_SUPPORTED_TEST_PLATFORMS=(
   linux/amd64
   linux/arm
   linux/arm64
-  linux/s390x
+  linux/mips64le
   linux/ppc64le
   darwin/amd64
   darwin/arm64
@@ -430,9 +430,9 @@ kube::golang::set_platform_envs() {
         export CGO_ENABLED=1
         export CC=${KUBE_LINUX_PPC64LE_CC:-powerpc64le-linux-gnu-gcc}
         ;;
-      "linux/s390x")
+      "linux/mips64le")
         export CGO_ENABLED=1
-        export CC=${KUBE_LINUX_S390X_CC:-s390x-linux-gnu-gcc}
+        export CC=${KUBE_LINUX_S390X_CC:-mips64el-linux-gnuabi64-gcc}
         ;;
     esac
   fi
-- 
2.25.1

